scripts/validate-schemas-final.js

```diff
@@ -0,0 +1,104 @@
+const fs = require('fs');
+const path = require('path');
+
+
+function validationDraft04(draft, startFileName, endFileName){
```

Comment:
This function seems a literal copy of the `validation` one. Could we just only use `validation` and remove this one?

Reply:
I divided into two function because of the difficulties faced between `draft-04` and `draft-07` validation.

I read [this line](https://ajv.js.org/json-schema.html#draft-04) from the ajv documentation for `draft-04` :point_right:   **`You can only use this import with JSON Schema draft-04, you cannot combine multiple JSON Schema versions in this ajv instance.`**

Since, we can only use one instance at once, I thought of making two function to validate the two different draft versions.

The `validationDraft04` starts from `1.0.0-without-$id.json` and ends on `1.2.0.json` because files ranging between these (and including those two) are of `draft-04` (I searched it in vscode). 

Hence, I made a separate function to validate `draft-04` files and since the new schemas from `2.0.0.json` are using `draft07`, I assumed future schemas will be written using `draft-7` or more latest draft versions.

For the `validation` function, it's range starts from `2.0.0-rc1.json` and it doesn't have a defined end point so if any new schema is being added, that too will be validated. It is having two exceptions `README.md` and `all.schema-store.json` namely.

So, if we don't want to validate the earlier schemas and only validate new incoming version or assuming newly developed schemas will have `draft-07` in them, we can remove the `validationDraft04` function.

WDYT?

Reply:
> Since, we can only use one instance at once, I thought of making two function to validate the two different draft versions.

But you are already instantiating one or another ajv based on the draft. See https://github.com/asyncapi/spec-json-schemas/pull/452/files#diff-1ee84b0b752624e8218489fead8e59e7dd93c1841a0b80418ecfa6cde1d57497R7

> The validationDraft04 starts from 1.0.0-without-$id.json and ends on 1.2.0.json because files ranging between these (and including those two) are of draft-04 (I searched it in vscode).

I agree @char0n's [comment](https://github.com/asyncapi/spec-json-schemas/pull/452#discussion_r1401755870). You can iterate over all schemas in `/schemas` and determine their metaschema (draft4 or draft7). You can check [the `$schema` field](https://github.com/asyncapi/spec-json-schemas/blob/master/schemas/1.0.0.json#L3) once you parse the file and you will get either `http://json-schema.org/draft-04/schema` or `http://json-schema.org/draft-07/schema`. Then you instantiate one or another ajv version. Does it sound reasonable?

Reply:
> You can iterate over all schemas in /schemas and determine their metaschema (draft4 or draft7). You can check [the $schema field](https://github.com/asyncapi/spec-json-schemas/blob/master/schemas/1.0.0.json#L3) once you parse the file and you will get either http://json-schema.org/draft-04/schema or http://json-schema.org/draft-07/schema. Then you instantiate one or another ajv version.

Okay, that will be a more optimized approach I think.

> Does it sound reasonable?

Yes, it does! I'll work on this approach

cc: @char0n Sorry, I missed [your comment above](https://github.com/asyncapi/spec-json-schemas/pull/452#discussion_r1401755870). The idea you've suggested would be more appropriate to implement. Thank you for your idea! :pray: 

***
> PS; _Sorry for the delayed response, running a bit hectic due to college exams :smiling_face_with_tear:_

Reply:
> > Since, we can only use one instance at once, I thought of making two function to validate the two different draft versions.
> 
> But you are already instantiating one or another ajv based on the draft. See https://github.com/asyncapi/spec-json-schemas/pull/452/files#diff-1ee84b0b752624e8218489fead8e59e7dd93c1841a0b80418ecfa6cde1d57497R7

Oh, sorry. I missed it :sweat_smile:

Reply:
implemented the idea through eccb4ab

marking as resolved